### PR TITLE
Connect redux to react component

### PR DESCRIPTION
**Pull request by oluwatobi-fasasi**
- Wrapped `App` with the `Provider` component.
- Passed configured store into the `Provider`.
- Created an initialState variable for book-Slice.
- Dispatched actions using useDispatch.
- Added an event handler that adds a book to the books array (with attributes id, title, and author)
- Added an event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 
 export default function Form() {
+  const dispatch = useDispatch();
   return (
     <div>
       <section className="mx-44 mt-11">
         <h2>ADD NEW BOOK</h2>
-        <form action="">
+        <form action="" onSubmit={() => dispatch(addBook({ title: '', author: '', id: '' }))}>
           <input type="text" className="border border-grey-900 h-10" />
```
Comment (OyePriscilla) [inline src/components/Form.js]: - [ ] Please kindly take note that I was unable to add a book to the list as expected, kindly fix this in order to proceed. 👍

File: src/components/BookStore.js [hunk 2]
```diff
@@ -1,24 +1,11 @@
-import React, { useEffect, useState } from 'react';
+// import React, { useEffect, useState } from 'react';
```
Comment (OyePriscilla) [inline src/components/BookStore.js]: - [ ] Kindly remove this commented code for your codebase to be clean. ✔️

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @oluwatobi-fasasi 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Used store, actions and reducers in React ✔️ 
Connected React and Redux ✔️ 
Implemented `Remove` functionality  ✔️ 

## Required Changes ♻️
**Please kindly amend the linkedln link that is broken at the moment, doing this will make your README to be more engaging.** 👍 
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Form.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+import { addBook } from '../redux/books/booksSlice';
 
 export default function Form() {
+  const dispatch = useDispatch();
+  const [author, setAuthor] = useState('');
+  const [title, setTitle] = useState('');
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    dispatch(addBook({ title, author, id: Date.now() }));
```
Comment (TedLivist) [inline src/components/Form.js]: - [ ] I was unable to remove a book I added. You need to use `item_id`, not `id` here.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @oluwatobi-fasasi,

✔️ Books can be added using local state

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
Hi @oluwatobi-fasasi  🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all the requirements ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**